dangerfile.js

```diff
@@ -16,15 +16,67 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import { schedule, danger, fail, warn, message, markdown } from "danger";
+// Removed import
 import fs from "fs";
 import path from 'path';
 import GitHubApi from 'github';
 import parseDiff from 'parse-diff';
 
+// check if pr submitted to master branch
+console.log("checkMasterBranch")
+const isMergeRefMaster = danger.github.pr.base.ref === 'master';
+if(!isMergeRefMaster){
+  fail("you must submit PR to master branch");
+}
+
+// match regex line by line
+function matchRegex(pr_body,regex){
+  const lines = pr_body.split("\n");
+  for (let i = 0; i < lines.length; i++) {
+    if(lines[i].match(regex)){
+      return true;
+    }
+  }
+  return false;
+}
+
+var pr_body = danger.github.pr.body.toLowerCase();
+// Because Pr description template include the following line：
+// 1. Write the corresponding [documentation](https://weex.io/guide/contribute/how-to-contribute.html#contribute-code-or-document)
+// so we should check the documentation below the ### checklist
+console.log("checkDocumentation");
+const index = pr_body.indexOf("checklist")
+const includeChecklist = (index!=-1)
+if(includeChecklist && !matchRegex(pr_body.substring(index),/documentation.*http/)){
+  const msg = "if you update the code, "+
+    "maybe you should update the documentation and add the documentation link in the PR description. \n" +
+    "here is the guide about how to contribute documentation:https://weex.apache.org/guide/contribute/how-to-contribute.html#contribute-code-or-document \n";
+  warn(msg);
+}
+
+// check if pr contains a demo link
+console.log("checkDemo");
+if(!matchRegex(pr_body,/demo.*http/)){
```

Comment:
You should allow both of the following spelling:
* demo
* Demo

Reply:
this is the same as above